### PR TITLE
Finalize the Capstone project.

### LINKED ISSUE #1: [1pt] #Group project: Create main.rb file
- [ ] Create a main.rb file that will serve as your console app entry-point.
- [ ] Implement startup actions:
  - Present the user with a list of options to perform.
  - Let users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.

### LINKED ISSUE #4: [1pt] #Group Task: Set-up project
- [ ] Set-up repository and add the collaborators.
- [ ] Create the general structure of the project.

### PR DESCRIPTION
**Pull request by M-Anwar-Hussaini**
#  🥇In the final version of Ruby Capstone Project. Here are the brief descriptions.
- [x] The program is a command-line-based project.
- [x] We have applied OOP concepts to our program.
- [x] We applied inheritance to the following classes, The `Item` is the parent and the others are the children.
  - `Book`
  - `Game`
  - `MusicAlbum`
  - `Movie`
  - `Item`
- [x] The `Item` class and its sub-classes are related to the following classes using an association (`composition` and `aggregation`) relationship.
  - `Author`
  - `Label`
  - `Source`
  - `Genre`
- [x] Apply unit tests for all classes.
- [x] Create the database structure for the program.
- [x] Preserving data in `.json`files.

## Important notes and Constraints.
- [x] An item type must have all its properties. it means that an instance of Item or its subclasses (`Book`, `Game`, `Movie`, `MusicAlbum`) must have `Genre`, `Source`, `Author `, and Label.
- [x] All item data is saved on `json/Itms.json` files.
- [x] We were 3 people but we did the activities of the 4th student as a bonus, that's why we didn't write unit tests for it.
## Video presentation
- [Here](https://www.youtube.com/watch?v=zlEN-XIbZO4) is the video link to describe shortly the project.

### CODE REVIEW

File: helpers/app.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      when 2 then add_genre
+      when 3 then add_label
+      when 4 then add_source
+      when 5 then add_item
+      when 6 then display_array(authors, 'List of authors: ')
+      when 7 then display_array(genres, 'List of genres: ')
+      when 8 then display_array(labels, 'List of labels: ')
+      when 9 then display_array(sources, 'List of sources: ')
+      when 10 then display_array(items, 'List of items: ')
+      when 0
```
Comment (DeliceLydia) [inline helpers/app.rb:108]: - [x] In this list you are missing options for adding
- Books
- Music album
- Game
- List all books
- List all music albums
- List of games
Please address that
Screenshot 2023-09-14 at 19 08 38

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Team, 

Good job so far! There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Highlights
- All members of the team participated in a video demo
- Tested all methods
- Kudos on working on the `sources` method even if it wasn't a requirement for a group of 3


## Suggested changes
_Check the comments under the review._



### Keep Rocking!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: helpers/app.rb (hunk 1 above)
Comment (M-Anwar-Hussaini) [reply helpers/app.rb:108]: Hi @DeliceLydia.
We have handled what you are saying in another way.

image


- [x] The number menu item contains adding all 4 items such as `Book`, `Music Album`, `Game`, and `Move` if there is at least one item for `Genre`, `Source`, `Author` and `Label`, It is a sub-menu of the 5th menu option.
- [x] The number 10 item displays all items from the 4 subclasses of `Items` such as  `Book`, `Music Album`, `Game`.

- Select to add one item:
image

- Add a book:
image

- Add a game:
image

- Add a movie:
image

- Add a music album
image

- List all items:
image

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @Team, 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Clean code
✅ ... Great readme file

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**